### PR TITLE
Fix release branch builds

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
Currently, when building Docker images, the ci-push-images.sh script would search for the latest hash in the same branch on the dashboard repo. The exception were tagged Kubermatic releases, which require a matching tag in the dashboard.

However, the dashboard does not build Docker images for untagged revisions in release branches. It only has images for each master branch revision and all tagged releases. This leads to cases where we take the latest revision from a dashboard release branch and then cannot find a Docker image for it.

This PR fixes that, hopefully.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/hack/lib.sh [hunk 1]
```diff
@@ -99,6 +99,30 @@ get_latest_dashboard_hash() {
...
+  local DEPTH="${2:-25}"
+
+  ensure_github_host_pubkey
+  git config --global core.sshCommand 'ssh -o CheckHostIP=no -i /ssh/id_rsa'
+  local DASHBOARD_URL="git@github.com:kubermatic/dashboard-v2.git"
+
+  local TMPDIR
+  TMPDIR=$(mktemp -d dashboard.XXXXX)
+
+  # git ls-remote cannot list tags in a meaningful way, so we have to clone the repo
```
Comment (kdomanski) [inline api/hack/lib.sh:115]: What does "list tags in a meaningful way" mean?

**Comment by kdomanski** [comment]
Any particular reason why you didn't just restore parts of the code removed in 1b9ac115b77944eabf0244720636915e72378ea5 ?

File: api/hack/lib.sh (hunk 1 above)
Comment (xrstf) [reply api/hack/lib.sh:115]: "meaningful" as in "you can tell which branch a tag belongs to", vs. just a flat list of all tags with no chance to determine the latest tag in a given branch.

**Comment by xrstf** [comment]
> Any particular reason why you didn't just restore parts of the code removed in 1b9ac11 ?

Because I did not search the history on this. I also think the old implementation removed in 1b9ac11 is inferior, as it seems to rely on string prefixes to match tags to branches.

**Comment by xrstf** [comment]
/retest

Fixed the CI system, the job should now actually work.

**Pull request merged**